Login Redesign & Mobile Login

theovier opened the pull request:
# Description

- #455 (partially)

## Reason for this PR
- Login page was not suitable on mobile devices

## Changes in this PR
- Login page for mobile devices

### Dependency update
- NPM: `"tailwind-heropatterns": "0.0.8",`

## Type of change
- Bug fix

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Manually tested all use cases this could alter

**Test Configuration**:
- OS: Windows Mac Linux
- Browser: Firefox Chrome Safari Chromium

- Frontend:
  - [x] Development build

- Backend:
  - [x] No backend needed to test this

# Checklist:

- [x] I have performed a self-review of my own code
- [x] My changes generate no new linting warnings or console warnings
- [ ] I have updated the CHANGELOG.md to include any changes made in this PR (add WIP to the top, if there is none already)

tobias-wiese reviewed (approved):
I think that is looking very nice and clean. I did some changes though:
- I disabled the login button based on the `isInputEmpty` computed (https://github.com/upb-uc4/ui-web/pull/698/commits/325e6c146b382c58f3ba019a6e572326a9b505ce feel free to revert)
- I moved the instantiation of the `LoginResponseHandler` into the method where it's used. I think this was a mistake we did earlier in the process (https://github.com/upb-uc4/ui-web/pull/698/commits/9cee01e8cd51d1522029bd8fa831f017c4c523e2)
- We do not have any `NotYetImplemented`-page that we could show for the 'Forgot your password?' thing. I implemented a suggestion of using our toasts, that are new on develop since yesterday. With that suggestion, just a warning toast with the message 'Not yet implemented.' is shown (see screenshot). This is just a suggestion so feel free to revert https://github.com/upb-uc4/ui-web/pull/698/commits/abcc4682e5acdccda2f61591d34f14f8ce4048bb if you do not like it :)
Screenshot:
![grafik](https://user-images.githubusercontent.com/63233799/99514667-4d103480-298c-11eb-904b-33940b590ea4.png)

theovier commented:
> I think that is looking very nice and clean. I did some changes though:
> 
> * I disabled the login button based on the `isInputEmpty` computed ([325e6c1](https://github.com/upb-uc4/ui-web/commit/325e6c146b382c58f3ba019a6e572326a9b505ce) feel free to revert)
> * I moved the instantiation of the `LoginResponseHandler` into the method where it's used. I think this was a mistake we did earlier in the process ([9cee01e](https://github.com/upb-uc4/ui-web/commit/9cee01e8cd51d1522029bd8fa831f017c4c523e2))
> * We do not have any `NotYetImplemented`-page that we could show for the 'Forgot your password?' thing. I implemented a suggestion of using our toasts, that are new on develop since yesterday. With that suggestion, just a warning toast with the message 'Not yet implemented.' is shown (see screenshot). This is just a suggestion so feel free to revert [abcc468](https://github.com/upb-uc4/ui-web/commit/abcc4682e5acdccda2f61591d34f14f8ce4048bb) if you do not like it :)
>   Screenshot:
>   ![grafik](https://user-images.githubusercontent.com/63233799/99514667-4d103480-298c-11eb-904b-33940b590ea4.png)

I like the changes. I thought about adding a not-implemented page too but thought that would not be in the scope of this PR. I really dig the idea of the alert.
